app/src/main/java/org/mozilla/focus/browser/VideoVoiceCommandMediaSession.kt

```diff
@@ -101,16 +101,16 @@ class VideoVoiceCommandMediaSession @UiThread constructor(
             addJavascriptInterface(JavascriptVideoPlaybackStateSyncer(), JS_INTERFACE_IDENTIFIER)
         }
 
-        val sessionIsLoadingObserver = SessionIsLoadingObserver(webView)
-        session.loading.observe(activity, sessionIsLoadingObserver)
+        val sessionIsLoadingObserver = SessionIsLoadingObserver(webView, session)
+        session.register(sessionIsLoadingObserver, owner = activity)
         this.sessionIsLoadingObserver = sessionIsLoadingObserver
     }
 
     fun onDestroyWebView(webView: IWebView, session: Session) {
         webView.removeJavascriptInterface(JS_INTERFACE_IDENTIFIER)
         this.webView = null
 
-        session.loading.removeObserver(sessionIsLoadingObserver!!)
+        sessionIsLoadingObserver?.let { session.unregister(it) }
```

Comment:
This should be non-null so I think we should assert that.

Reply:
👍